src/test/java/core/basesyntax/RegistrationServiceTest.java

```
@@ -0,0 +1,143 @@
+    }
+
+    @Test
+    public void testLogin_WithTooShortLogin_NotOk() {
+        User user = new User("asdfg", "poi5437", 64);
+        User user1 = new User("kutam76", "1234tyr", 42);
+        User user2 = new User("tyre3241", "9843oitr", 24);
+        RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+        registrationService.register(user1);
+        registrationService.register(user2);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void testLogin_WithNullLogin_NotOk() {
+        User user = new User(null, "098756hg", 54);
+        User user1 = new User("jhere123", "74568erw", 33);
+        User user2 = new User("troi4576", "984321ed", 31);
+        RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+        registrationService.register(user1);
+        registrationService.register(user2);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void testLogin_WithBlankLogin_NotOk() {
+        User user = new User("", "ptyre123", 20);
+        User user1 = new User("rtjww23", "rewku234", 44);
+        User user2 = new User("tyrex45", "12384fhg", 33);
+        RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+        registrationService.register(user1);
+        registrationService.register(user2);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void testLogin_IfUserWithGivenLoginExists_NotOk() {
+        User user = new User("qwerty23", "poty3456", 21);
+        User user1 = new User("lowysh432", "portugal1283", 45);
+        User user2 = new User("qwerty23", "poty3456", 21);
+        RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+        registrationService.register(user1);
+        registrationService.register(user2);
```

Issue: In this test, you are registering `user2` with the same login as `user` before checking the exception. According to your registration logic, registering `user2` should already throw an `InvalidDataException` due to the duplicate login. You should only register one user with the login, then assert that registering another with the same login throws the exception.